CorsixTH/Lua/config_finder.lua

```diff
@@ -493,4 +502,317 @@ audio_music = nil -- [[X:\ThemeHospital\Music]]
+  ingame_storePosition_0 = {"alt", "0"},
+  ingame_recallPosition_1 = {"ctrl", "1"},
+  ingame_recallPosition_2 = {"ctrl", "2"},
+  ingame_recallPosition_3 = {"ctrl", "3"},
+  ingame_recallPosition_4 = {"ctrl", "4"},
+  ingame_recallPosition_5 = {"ctrl", "5"},
+  ingame_recallPosition_6 = {"ctrl", "6"},
+  ingame_recallPosition_7 = {"ctrl", "7"},
+  ingame_recallPosition_8 = {"ctrl", "8"},
+  ingame_recallPosition_9 = {"ctrl", "9"},
+  ingame_recallPosition_0 = {"ctrl", "0"},
+}
+
+-- Clear the loaded file variable.
+fi = nil
+fi = io.open(hotkeys_filename, "r")
+local hotkeys_values = {}
+local hotkeys_needs_rewrite = false
+for key, value in pairs(hotkeys_defaults) do
+  hotkeys_values[key] = value
+end
+
+-- If the file opened succesfully...
+if fi then
+
+  local file_contents = fi:read("*all")
+  fi:close()
+  local temp_string_01 = ""
+
+  for key, value in pairs(hotkeys_defaults) do
+    local ind = string.find(file_contents, "\n" .. "%s*" .. key .. "%s*=")
+	-- If we couldn't find the key in the hotkeys.txt file...
+    if not ind then
+      hotkeys_needs_rewrite = true
+	-- Otherwise, if the key was found in the hotkeys.txt file...
+    else
+      ind = ind + (string.find(file_contents, key, ind) - ind) + string.len(key)
+      ind = string.find(file_contents, "=", ind) + 1
+      
+      -- If the value of the key we are currently looking at is NOT a string...
+      if type(value) ~= "string" then
+        ind = string.find(file_contents, "[%a%d]", ind)
+
+        -- If the value to we read or written is a table...
+		if type(value) == "table" then
+		  hotkeys_values[key] = value
+      	else
+          hotkeys_values[key] = string.sub(file_contents, ind, string.find(file_contents, "[ \n-]", ind + 1) - 1)
+        end
+      -- Otherwise, if the value we are looking for IS a string...
+      else
+        ind = string.find( file_contents, "[", ind + 1, true ) + 1
+
+		hotkeys_values[key] = string.sub(file_contents, ind + 1, string.find(file_contents, "]", ind, true) - 1)
+      end
+    end
+  end
+else
+  hotkeys_needs_rewrite = true
+end
+
+--
+if hotkeys_needs_rewrite then
+  -- The config file that will be written is divided into seperate strings, which are concatenated when they are written into the file.
+  -- This is done to avoid a "Too many C levels" error.
+  local string_01 = [=[
+----------------------------------------CorsixTH Hotkey Mappings File----------------------------------------
+-- Lines starting with two dashes (like this one) are ignored.
+-- Text settings should have their values between double square braces, e.g.
+-- Number settings should not have anything around their value,
+--  setting = [[value]]
+-- e.g. setting = 42
+
+----------------------------------------Global Keys----------------------------------------
+-- These are global keys to be used at anytime while the application is open.
+--]=] .. '\n' ..
+'global_confirm = [[' .. tostring(hotkeys_defaults.global_confirm) .. ']]' .. '\n' ..
```

Comment:
All of these should support hotkeys with modifiers. I can put that in my patch if you want.

Reply:
Oh, yeah. If you'd like, you can.

Reply:
It ended up being bigger than I anticipated. A simple array join isn't suitable for the config and I'm ending up writing a general purpose serializer based loosely on the old print_table.

Maybe a few days due to my availability.

Reply:
No problem. Take your time. I'll work on testing and a couple other things in the meantime.